api/repositories/security_group_repository_test.go

```diff
@@ -0,0 +1,78 @@
+package repositories_test
+
+import (
+	apierrors "code.cloudfoundry.org/korifi/api/errors"
+	"code.cloudfoundry.org/korifi/api/repositories"
+	korifiv1alpha1 "code.cloudfoundry.org/korifi/controllers/api/v1alpha1"
+	"code.cloudfoundry.org/korifi/tests/matchers"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("SecurityGroupRepo", func() {
+	var (
+		repo  *repositories.SecurityGroupRepo
+		org   *korifiv1alpha1.CFOrg
+		space *korifiv1alpha1.CFSpace
+	)
+
+	BeforeEach(func() {
+		repo = repositories.NewSecurityGroupRepo(userClientFactory, rootNamespace)
+		org = createOrgWithCleanup(ctx, prefixedGUID("org"))
+		space = createSpaceWithCleanup(ctx, org.Name, prefixedGUID("space"))
+	})
+
+	Describe("CreateSecurityGroup", func() {
+		var (
+			securityGroupRecord        repositories.SecurityGroupRecord
+			securityGroupCreateMessage repositories.CreateSecurityGroupMessage
+			createErr                  error
+		)
+
+		BeforeEach(func() {
+			securityGroupCreateMessage = repositories.CreateSecurityGroupMessage{
+				DisplayName: "test-security-group",
+				Rules: []korifiv1alpha1.SecurityGroupRule{
+					{
+						Protocol:    korifiv1alpha1.ProtocolTCP,
+						Ports:       "80",
+						Destination: "192.168.1.1",
+					},
+				},
+				Spaces: map[string]korifiv1alpha1.SecurityGroupWorkloads{
+					space.Name: {Running: true, Staging: true},
+				},
+			}
+		})
+
+		JustBeforeEach(func() {
+			securityGroupRecord, createErr = repo.CreateSecurityGroup(ctx, authInfo, securityGroupCreateMessage)
+		})
+
+		It("errors with forbidden for users with no permissions", func() {
+			Expect(createErr).To(matchers.WrapErrorAssignableToTypeOf(apierrors.ForbiddenError{}))
+		})
+
+		When("the user is a CF admin", func() {
+			BeforeEach(func() {
+				createRoleBinding(ctx, userName, adminRole.Name, rootNamespace)
+			})
+
+			It("creates a CFSecurityGroup successfully", func() {
```

Comment:
We also want to check that the `Create` call also created expected `CFSecurityGroup` object in kubernetes, maybe in another `It`